### ISSUE TITLE
There is a problem with initial-page props and gotoPage method

### ISSUE THREAD

**Issue by lrembacz** [ratiw/vuetable-2#469]
Hi, I want to bring back the state from my storage after refresh the page.
So I actually make the vueTable initOnStart prop to false and then after I get all my storage information I just want to loadData().

Ive noticed the bug. 
When Vuetable is initialized its initalPage state is taken from props and passed to data directly with:
```
currentPage: this.initialPage,
```
So after we want to change this prop or just change initial page we would need to destroy this instance and re-render it again to get the state from props. Especially when we dynamically changed this props with Vuex or something.

GoToPage method cannot be actually used for init this current page because without loading the state we have no last_page property on `tablePagination`. `"TypeError: Cannot read property 'last_page' of null"`
```
gotoPage (page) {
  if (page != this.currentPage && (page > 0 && page <= this.tablePagination.last_page)) {
    this.currentPage = page
    this.loadData()
  }

```

First init state && watcher for the initialPage would be good solution. Or method where we can pass this currentPage property. Then we would need to check if there is page given in our response from server. And if its not just change currentPage to last_page or something.

Doing second call just to change the page is not best solution I guess... especially if we are making call on bigger DB.

**Comment by ratiw** [comment]
@lrembacz I guess you're using Data mode (`:api-mode="false"`) together with `VuetablePagination` component, am I right?

**Comment by lrembacz** [comment]
@ratiw No I am actually using :api-mode="true".

Check my JSFiddle example with same action. Passing props as data, its not reactive. 

https://jsfiddle.net/4x5t7Lnw/

You would need watcher actually.

https://jsfiddle.net/4x5t7Lnw/2/

**Comment by ratiw** [comment]
@lrembacz I see. Adding the watcher is easy enough. However, this may not be sufficient as others might require other props as well and this will lead to adding watcher to every prop. 

How about if Vuetable provides a hook function, for example `loadState`, that will be called when it finishes initializing (but before starting loading data). This function will receive an object called `state` containing some data (e.g. `currentPage`, `tableFields` etc.) that you could change and return back, and Vuetable will use those changes to override its data variables. The only problem is that this will work on variables in the `data` section since the prop cannot be changed by the component itself.

**Comment by lrembacz** [comment]
@ratiw Yes, watcher was just example but it's not good enough in this solution.

The problem about initialPage is even much deeper. This problem exists for every data/prop used once.
Unfortunately I don't know the pattern for this solution and can't see any on documentation page.
Initial page prop is actually used once so I  think we don't need to change it reactively.  

For now I just directly set this `currentPage` before `loadData()` with 
```
this.$refs.vuetable.currentPage = this.initialPage;
```
But its not good as well.

Maybe just additional data variable, like `inited`  would be the option here as well. So this data state would determinate if we were taking `currentPage` or `initialPage` for first call.

For now I think method to change the sate programmatically is good solution I guess. Or just the set of setters to change some initial data before load from DB.
